mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -453,16 +454,17 @@ class WidgetAdapter(
         initData: ViewHolderInitData,
         @LayoutRes layoutResId: Int,
         @LayoutRes compactModeLayoutResId: Int = layoutResId
-    ) : ViewHolder(initData, layoutResId, compactModeLayoutResId) {
-        protected var boundWidget: Widget? = null
-            private set
+    ) : LabeledItemBaseViewHolder(initData, layoutResId, compactModeLayoutResId) {
         protected val widgetContentView: View = itemView.findViewById(R.id.widget_content)
         private val dataSaverView: View = itemView.findViewById(R.id.data_saver)
         private val dataSaverButton: Button = itemView.findViewById(R.id.data_saver_button)
         private val dataSaverHint: TextView = itemView.findViewById(R.id.data_saver_hint)
 
         override fun bind(widget: Widget) {
-            boundWidget = widget
+            super.bind(widget)
+            val showLabelAndIcon = widget.label.isNotEmpty()
```

Comment:
Show the label and icon if the label is empty or if label and icon are empty?

Reply:
Keep it consistent with Basic UI: Label must be empty to hide the label and icon.